connection_generator.py

```diff
@@ -0,0 +1,197 @@
+"""
+This module implements methods for generating random connections between nodes in a graph.
+in a neo4j database.
+
+Method generate() will create all the necessary connections for the graph:
+    dataset <-> dataset collection
+    system <-> system collection
+    dataset read <-> system input
+    dataset write <-> system output
+"""
+
+from itertools import islice
+import random
+
+
+class ConnectionGenerator:
+    """
+    A class to generate random connections between node ids, based on distribution maps.
+
+    ...
+
+    Attributes:
+        dataset_count: Integer of how many datasets are in a graph.
+        dataset_count_map: Dictionary that maps number of datasets in collection to count of dataset collections.
```

Comment:
As there's no test, consider adding type for dictionaries (e.g. int:int or int:[int]) for the reader to better understand